ts/change-notetype/MapperRow.svelte

```diff
@@ -21,29 +21,34 @@ License: GNU AGPL, version 3 or later; http://www.gnu.org/licenses/agpl.html
         current = oldNames[newIndex] || tr.changeNotetypeNothing();
     }
 
-    function onChange(evt: Event) {
-        const oldIdx = parseInt((evt.target as HTMLSelectElement).value, 10);
-        state.setOldIndex(ctx, newIndex, oldIdx);
-    }
-
     // optimization for big notetypes
     let active: boolean = false;
     function activate(evt: Event) {
         active = true;
     }
+
+    function onChange(evt: Event) {
+        const oldIdx = parseInt((evt.target as HTMLSelectElement).value, 10);
+        state.setOldIndex(ctx, newIndex, oldIdx);
+    }
 </script>
 
 <Row --cols={2}>
     <Col --col-size={1}>
         <!-- svelte-ignore a11y-no-onchange -->
         <select
-            value={$info.getOldIndex(ctx, newIndex)}
+            value={active ? $info.getOldIndex(ctx, newIndex) : 0}
             class="form-select"
+            on:focusin={activate}
             on:change={onChange}
         >
-            {#each $info.getOldNamesIncludingNothing(ctx) as name, idx}
-                <option value={idx}>{name}</option>
-            {/each}
+            {#if active}
+                {#each oldNames as name, idx}
+                    <option value={idx}>{name}</option>
+                {/each}
+            {:else}
+                <option value={0}>{current}</option>
+            {/if}
```

Comment:
By only loading the selected `option` and loading the rest on focus, we can significantly reduce load times and window resize lag.

Say we have 500 fields and 500 templates, instead of 2*500² = 500.000 options, we only load 1000.

Reply:
Nice!

Reply:
... but unfortunately this does not appear to work on mobile - only a single option is shown when tapping on a dropdown, until it is tapped again. Maybe the focus event is not firing quick enough? You can test locally with a mobile device using ts-run - if you're an Android user, it's possible the same issue exists in mobile Chrome: https://github.com/ankitects/anki/issues/1174

Reply:
// use #testXXXX where XXXX is notetype ID to test

Reply:
That's a bummer :/ My phone cable is a bit loose, so I couldn't get mobile debugging to reliably work right now. I gotta go to bed now, but I'll test that tomorrow.

Generally, writing a `Lazy` Svelte component might be a good idea, since the editor shows similar performance issues :thinking: